tests/avro/test_avro_producer.py

```diff
@@ -20,57 +20,66 @@
 # derived from https://github.com/verisign/python-confluent-schemaregistry.git
 #
 import os
-import sys
 
 from confluent_kafka import avro
 
-if sys.version_info[0] < 3:
-    import unittest
-else:
-    import unittest2 as unittest
+import unittest
 from confluent_kafka.avro import AvroProducer
+from confluent_kafka.avro.serializer import SerializerError
 
 avsc_dir = os.path.dirname(os.path.realpath(__file__))
 
 
 class TestAvroProducer(unittest.TestCase):
-    def setUp(self):
-        pass
 
     def test_instantiation(self):
         obj = AvroProducer({'schema.registry.url': 'http://127.0.0.1:0'})
         self.assertTrue(isinstance(obj, AvroProducer))
         self.assertNotEqual(obj, None)
 
-    def test_Produce(self):
-        producer = AvroProducer({'schema.registry.url': 'http://127.0.0.1:0'})
-        valueSchema = avro.load(os.path.join(avsc_dir, "basic_schema.avsc"))
+    def test_produce_no_key(self):
+        value_schema = avro.load(os.path.join(avsc_dir, "basic_schema.avsc"))
+        producer = AvroProducer({'schema.registry.url': 'http://127.0.0.1:9001'}, default_value_schema=value_schema)
         try:
-            producer.produce(topic='test', value={"name": 'abc"'}, value_schema=valueSchema, key='mykey')
-            self.fail("Should expect key_schema")
-        except Exception as e:
+            # No key should work
+            producer.produce(topic='test', value={"name": 'abc"'})
+        except SerializerError:
+            self.fail("Produce should accept empty keys")
```

Comment:
Based on your other comments below I'm in the mind that we should indeed break up the key and value serialization into different methods and raise two different exceptions: `KeySerializerError` and `ValueSerializerError`. This will help tune these test better.